bin/api-docs/gen-theme-reference.js

```diff
@@ -127,16 +127,44 @@ const getSettingsPropertiesMarkup = ( struct ) => {
 	}
 
 	let markup = '| Property  | Type   | Default | Props  |\n';
-	markup += '| ---       | ---    | ---    |---   |\n';
+	markup += '| ---    | ---    | ---    |---   |\n';
 	ks.forEach( ( key ) => {
 		const def = 'default' in props[ key ] ? props[ key ].default : '';
-		const ps =
+		let type = props[ key ].type || '';
+		let ps =
 			props[ key ].type === 'array'
 				? keys( getPropertiesFromArray( props[ key ].items ) )
 						.sort()
 						.join( ', ' )
 				: '';
-		markup += `| ${ key } | ${ props[ key ].type } | ${ def } | ${ ps } |\n`;
+
+		/*
+		 * Handle`oneOf` type definitions - extract the type and properties.
+		 * See: https://json-schema.org/understanding-json-schema/reference/combining#oneOf
+		 */
+		if ( props[ key ].oneOf && Array.isArray( props[ key ].oneOf ) ) {
+			if ( ! type ) {
+				type = props[ key ].oneOf
+					.map( ( item ) => item.type )
+					.join( ', ' );
+			}
+
+			if ( ! ps ) {
+				ps = props[ key ].oneOf
+					.map( ( item ) =>
+						item?.type === 'object' && item?.properties
+							? '_{' +
```

Comment:
I did this to group properties where there might be multiple object types in a single `"oneOf"`.

Open to suggestions.

🤷🏻

Reply:
Might be outside the scope of this PR, but wouldn't it be possible to apply this format to style properties as well? I think this can be resolved by making the following changes.

```diff
diff --git a/bin/api-docs/gen-theme-reference.js b/bin/api-docs/gen-theme-reference.js
index 6bb0e5d5b4..762412f762 100644
--- a/bin/api-docs/gen-theme-reference.js
+++ b/bin/api-docs/gen-theme-reference.js
@@ -191,7 +191,9 @@ const getStylePropertiesMarkup = ( struct ) => {
        ks.forEach( ( key ) => {
                const ps =
                        props[ key ].type === 'object'
-                               ? keys( props[ key ].properties ).sort().join( ', ' )
+                               ? '_{' +
+                                 keys( props[ key ].properties ).sort().join( ', ' ) +
+                                 '}_'
                                : '';
                const type = formatType( props[ key ] );
                markup += `| ${ key } | ${ type } | ${ ps } |\n`;
```